### PR TITLE
Addition of devise gem and devise generation

### PR DESCRIPTION
**Pull request by LoviMadaan**
### In this milestone, We have followed the pair-programming approach and done the following:
______________________________________________________________________________________________________________________________________
**Project requirements**

- [ ] Remove current_user method in ApplicationController, devise will provide us with one.
- [ ] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- [ ] Can register a new user.
- [ ] User logs in with a combination of email and password.
- [ ] Hashed password should be stored in the database.
- [ ] Ask for confirmation of email.
- [ ] Can reset password.
- [ ] [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @LoviMadaan 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project, that has failed linter checks. Kindly check your styling errors and fix and submit a project that has linters passing.🙏🏾

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,15 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Owusu-Desmond) [inline app/models/user.rb:5]: - [ ] Please from the project requirement, You should enable `:confirmable` in your user model to ask for confirmation of email after sign up to ensure that the user email is confirmed before they complete the signup process :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @LoviMadaan , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,15 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (Lizdev-05) [inline app/models/user.rb:5]: - [ ] As requested by the previous reviewer, you should ensure that a user email is confirmed before the signup is complete but here is the case if I fill in the signup form and click on the signup button, I get the error below :point_down:  so make sure that you fix this and ensure the confirmation of email works perfectly :+1: 

  image

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @LoviMadaan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (LoviMadaan) [reply app/models/user.rb:5]: @Lizdev-05 can you explain how can i resolve this error.. i tried to resolve but i am not able to..

Comment (LoviMadaan) [reply app/models/user.rb:5]: Bug fixed 
Screenshot 2023-07-21 152227

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @LoviMadaan, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights :heavy_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Implemented some of the changes from the previous reviewer 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi @LoviMadaan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Great follow-up with the previous reviewers
- All functionality working as expected.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**